lib/pbench/agent/tool_meister.py

```diff
@@ -200,210 +347,141 @@ def stop(self):
             self.tool_opts,
         ]
         self.logger.info("%s: stop_tool -- %s", self.name, " ".join(args))
-        o_file = self.tool_dir / f"tm-{self.name}-stop.out"
-        e_file = self.tool_dir / f"tm-{self.name}-stop.err"
-        with o_file.open("w") as ofp, e_file.open("w") as efp:
-            self.stop_process = subprocess.Popen(
-                args,
-                cwd=self.tool_dir,
-                stdin=subprocess.DEVNULL,
-                stdout=ofp,
-                stderr=efp,
-            )
-
-    def _wait_for_process(self, process):
-        """_wait_for_process - generic method to wait for a given process to
-        stop after 30 seconds, emitting a message every 5 seconds in between.
-
-        Returns the process return code on success, or None if the process
-        failed to exit.
-        """
-        count = 6
-        sts = None
-        while count > 0 and sts is None:
-            try:
-                sts = process.wait(timeout=5)
-            except subprocess.TimeoutExpired:
-                count -= 1
-                if count > 0:
-                    self.logger.info(
-                        "%s tool %s has not stopped after %d seconds",
-                        self._tool_type,
-                        self.name,
-                        5 * (6 - count),
-                    )
-        return sts
+        self.stop_process = self._create_process_with_logger(
+            args, self.tool_dir, "stop"
+        )
 
     def wait(self):
         """Wait for any tool processes to terminate after a "stop" process has
         completed.
 
         Waits for the tool's "stop" process to complete, if started, then
-        waits for the tool's start process to complete.
+        waits for the tool's start process to complete (since the "stop"
+        process is supposed stop the "start" process).
         """
+        assert self.tool_dir is not None, "Logic bomb!  no tool directory provided!"
+        if not self.tool_dir.is_dir():
+            raise RuntimeError(f"tool directory does not exist: {self.tool_dir}")
```

Comment:
Because until PR #2697 lands, `self.tool_dir` must be defined.  Meaning, `.wait()` should only be called in a context where `self.tool_dir` is present.

Reply:
This probably isn't worth arguing over, but I fail to see how it matters to _this function_ whether that attribute has a value, since this function doesn't reference it.  It seems like, if we're trying to protect a subroutine or otherwise verify an invariant, that check should either be in the code which needs protection or at some level above this one.

(I totally get the idea of sprinkling the flow with debugging statements to try to figure out where something goes south, but this is a PR, so the code should have been debugged by now, and, to the extent that we need to retain these checks, they should be properly positioned...and I don't yet see why this is "the right place" for these.)

Put another way, is a `RuntimeError` from here really better than an `AttributeError` from somewhere deeper in the stack, particularly for a case that we don't expect to occur?